pkg/adapter/reconciler/blueprint_controller.go

```diff
@@ -40,29 +42,37 @@ func (r *BlueprintReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	err := r.blueprintChangeHandler.HandleChange(ctx, req.Name)
 
+	if err != nil {
+		return decideRequeueForError(logger, err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func decideRequeueForError(logger logr.Logger, err error) (ctrl.Result, error) {
+	errLogger := logger.WithValues("error", err)
+
 	var internalError *domainservice.InternalError
 	var conflictError *domainservice.ConflictError
 	var notFoundError *domainservice.NotFoundError
-	var invalidError *domain.InvalidBlueprintError
-	if err != nil {
-		logger := logger.WithValues("error", err)
-		if errors.As(err, &internalError) {
-			logger.Error(err, "An internal error occurred and can maybe be fixed by retrying it later")
-			return ctrl.Result{}, err // automatic requeue
-		} else if errors.As(err, &conflictError) {
-			logger.Info("A concurrent update happened in conflict to the processing of the blueprint spec. A retry could fix this issue")
-			return ctrl.Result{Requeue: true, RequeueAfter: 1 * time.Second}, nil // no error as this would lead to the ignorance of our own retry params
-		} else if errors.As(err, &notFoundError) {
-			logger.Info("blueprint was not found, so maybe it was deleted in the meantime. No further evaluation will happen")
-			return ctrl.Result{Requeue: false}, nil
-		} else if errors.As(err, &invalidError) {
-			logger.Info("blueprint is invalid, therefore there will be no further evaluation.")
-			return ctrl.Result{Requeue: false}, nil
-		}
-		logger.Error(err, "an unknown error type occurred. Retry with default backoff")
-		return ctrl.Result{}, err // automatic requeue
+	var invalidBlueprintError *domain.InvalidBlueprintError
+	switch {
+	case errors.As(err, &internalError):
+		errLogger.Error(err, "An internal error occurred and can maybe be fixed by retrying it later")
+		return ctrl.Result{}, err //automatic requeue because of non-nil err
+	case errors.As(err, &conflictError):
+		errLogger.Info("A concurrent update happened in conflict to the processing of the blueprint spec. A retry could fix this issue")
+		return ctrl.Result{Requeue: true, RequeueAfter: 1 * time.Second}, nil // no error as this would lead to the ignorance of our own retry params
+	case errors.As(err, &notFoundError):
+		errLogger.Info("Blueprint was not found, so maybe it was deleted in the meantime. No further evaluation will happen")
+		return ctrl.Result{Requeue: false}, nil
+	case errors.As(err, &invalidBlueprintError):
+		errLogger.Info("Blueprint is invalid, therefore there will be no further evaluation.")
+		return ctrl.Result{Requeue: false}, nil
+	default:
+		errLogger.Error(err, "An unknown error type occurred. Retry with default backoff")
+		return ctrl.Result{}, err //automatic requeue because of non-nil err
 	}
```

Comment:
Nicely done! I didn't know that you could use `switch` instead of multiple `if - else if` statements.